qa/rpc-tests/walletbackup.py

```diff
@@ -133,9 +133,21 @@ def run_test(self):
         for i in range(5):
             self.do_one_round()
 
-        # Generate 101 more blocks, so any fees paid mature
-        self.nodes[3].generate(101)
-        self.sync_all()
+        print "Generate 101 more blocks, so any fees paid mature"
+        #separating the block creation in this step is to get around issue #6554 which happens on Windows
+        if os.getenv('EXEEXT', "") == '.exe': 
+            print "WARNING: mining smaller numbers of blocks in this step to workaround issue #6554"
+            self.nodes[3].generate(25)
+            self.sync_all()
+            self.nodes[3].generate(25)
+            self.sync_all()
+            self.nodes[3].generate(25)
+            self.sync_all()
+            self.nodes[3].generate(26)
+            self.sync_all()
+        else:
+            self.nodes[3].generate(101)
+            self.sync_all()
```

Comment:
i hear you, but there's no other way around for the time being until 
issue 6554 is fixed.  Once fixed we can take out this section of code.

On 17/08/2015 8:30 AM, Jonas Schnelli wrote:
>
> In qa/rpc-tests/walletbackup.py 
> <https://github.com/bitcoin/bitcoin/pull/6548#discussion_r37199876>:
>
> > -        self.sync_all()
> > +        print "Generate 101 more blocks, so any fees paid mature"
> > +        #separating the block creation in this step is to get around issue #6554 which happens on Windows
> > +        if os.getenv('EXEEXT', "") == '.exe':
> > +            print "WARNING: mining smaller numbers of blocks in this step to workaround issue #6554"
> > +            self.nodes[3].generate(25)
> > +            self.sync_all()
> > +            self.nodes[3].generate(25)
> > +            self.sync_all()
> > +            self.nodes[3].generate(25)
> > +            self.sync_all()
> > +            self.nodes[3].generate(26)
> > +            self.sync_all()
> > +        else:
> > +            self.nodes[3].generate(101)
> > +            self.sync_all()
>
> This change feels wrong. Platform specific code in tests and Github 
> issue references in the source code should be avoided.
> Is there no way to fix the root cause of this issue? If we can't sync 
> >25 blocks on windows then there is a issue which needs fixing.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/bitcoin/bitcoin/pull/6548/files#r37199876>.
>